celext/lib.go

```diff
@@ -467,3 +486,36 @@ func (l lib) validateIPPrefix(p string, ver int64, strict bool) bool {
 		return false
 	}
 }
+
+func (l lib) isHostAndPort(val string, portRequired bool) bool {
+	splitIdx := strings.LastIndexByte(val, ':')
+
+	if val[0] == '[' { // ipv6
```

Comment:
Should this verify the string isn't empty first?

Reply:
oh yes for sure 😬